src/com/ms/silverking/cloud/zookeeper/ZooKeeperExtended.java

```diff
@@ -42,1024 +43,1081 @@
-                while (_zk.getState() == States.CONNECTING) {
-                    ThreadUtil.sleep(connectionCheckIntervalMillis);
-                }
-                break;
-            } catch (IOException ioe) {
-                Log.logErrorWarning(ioe);
-                if (curAttempt >= connectAttempts) {
-                    throw ioe;
-                }
-            } catch (KeeperException ke) {
-                Log.logErrorWarning(ke);
-                if (curAttempt >= connectAttempts) {
-                    throw ke;
-                }
-            }
-            curAttempt++;
-            ThreadUtil.randomSleep(0, (1 << curAttempt) * 1000);
-        } while (curAttempt < connectAttempts);
-        return _zk;
+  }
+
+  public List<String> getChildren(String path) throws KeeperException {
+    try {
+      return getChildren(path, false);
+    } catch (InterruptedException ie) {
+      throw new RuntimeException("panic");
     }
-    
-    public void createAllNodes(Collection<String> paths) throws KeeperException {
-        for (String path : paths) {
-            if (path != null) {
-                createAllNodes(path);
-            }
-        }
+  }
+
+  public List<String> getChildrenPriorTo(String path, long zxid) throws KeeperException {
+    List<String> allChildren;
+    List<String> priorChildren;
+
+    allChildren = getChildren(path);
+    priorChildren = new ArrayList<>();
+    for (String child : allChildren) {
+      //System.out.println("\t"+ child);
+      //System.out.println("\t\t"+ getStat(path +"/"+ child).getMzxid() +"\t"+ zxid);
+      if (getStat(path + "/" + child).getMzxid() < zxid) {
+        priorChildren.add(child);
+      }
+    }
+    return priorChildren;
+  }
+
+  public Stat getStat(String path) throws KeeperException {
+    try {
+      Stat stat;
+
+      stat = new Stat();
+      getData(path, false, stat);
+      return stat;
+    } catch (InterruptedException ie) {
+      throw new RuntimeException("panic");
     }
-    
-    public void createAllParentNodes(String path) throws KeeperException {
-        int index;
-        
-        if (!path.startsWith("/")) {
-            throw new RuntimeException("Bad path:"+ path);
-        }
-        index = path.lastIndexOf('/');
-        if (index >= 0) {
-            createAllNodes(path.substring(0, index));
-        } else {
-            // root level; no creation necessary
-        }
+  }
+
+  public ChildrenCallbackOp getChildrenAsync(String path, boolean watch) throws KeeperException {
+    ChildrenCallbackOp op;
+
+    op = newChildrenCallbackOp();
+    super.getChildren(path, watch, (ChildrenCallback) this, op.getRequestUUID());
+    return op;
+  }
+
+  public ChildrenCallbackOp getChildrenAsync(String path) throws KeeperException {
+    return getChildrenAsync(path, false);
+  }
+
+  public void close() {
+    try {
+      super.close();
+    } catch (InterruptedException ie) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)